Gems/Atom/RHI/Code/Include/Atom/RHI/MultiDeviceSwapChain.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <Atom/RHI.Reflect/SwapChainDescriptor.h>
+#include <Atom/RHI/MultiDeviceImagePoolBase.h>
+#include <Atom/RHI/SwapChain.h>
+#include <Atom/RHI/XRRenderingInterface.h>
+
+namespace AZ::RHI
+{
+    //! The platform-independent, multi-device swap chain base class. Swap chains contain a "chain" of images which
+    //! map to a platform-specific window, displayed on a physical monitor. The user is allowed
+    //! to adjust the swap chain outside of the current FrameScheduler frame. Doing so within a frame scheduler
+    //! frame results in undefined behavior.
+    //!
+    //! The frame scheduler controls presentation of the swap chain. The user may attach a swap chain to a scope
+    //! in order to render to the current image.
+    class MultiDeviceSwapChain : public MultiDeviceImagePoolBase
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(MultiDeviceSwapChain, AZ::SystemAllocator, 0);
+        AZ_RTTI(MultiDeviceSwapChain, "{EB2B3AE5-41C0-4833-ABAD-4D964547029C}", Object);
+        MultiDeviceSwapChain() = default;
+        virtual ~MultiDeviceSwapChain() = default;
+
+        //! Initializes the swap chain, making it ready for attachment.
+        ResultCode Init(MultiDevice::DeviceMask deviceMask, const SwapChainDescriptor& descriptor);
+
+        //! Returns the device-specific SwapChain for the given index
+        inline Ptr<SwapChain> GetDeviceSwapChain(int deviceIndex) const
+        {
+            AZ_Error(
+                "MultiDeviceSwapChain",
+                m_deviceSwapChains.find(deviceIndex) != m_deviceSwapChains.end(),
+                "No DeviceSwapChain found for device index %d\n",
+                deviceIndex);
+
+            return m_deviceSwapChains.at(deviceIndex);
+        }
+
+        //! Presents the swap chain to the display, and rotates the images.
+        void Present();
+
+        //! Sets the vertical sync interval for the swap chain.
+        //!      0 - No vsync.
+        //!      N - Sync to every N vertical refresh.
+        //!
+        //! A value of 1 syncs to the refresh rate of the monitor.
+        void SetVerticalSyncInterval(uint32_t verticalSyncInterval);
+
+        //! Resizes the display resolution of the swap chain. Ideally, this matches the platform window
+        //! resolution. Typically, the resize operation will occur in reaction to a platform window size
+        //! change. Takes effect immediately and results in a GPU pipeline flush.
+        ResultCode Resize(const SwapChainDimensions& dimensions);
+
+        //! Returns the number of images in the swap chain.
+        uint32_t GetImageCount() const;
+
+        //! Returns the current image of the swap chain.
```

Comment:
Add comment about which device's swap chain it is returning.

Reply:
This function is returning a `MultiDeviceImage*`, so it is not clear at this point. Most likely the `MultiDeviceSwapChain` will only be initialized for a single device, but then we still return a `MultiDeviceImage*` which then holds one `Image`.